### PR TITLE
Api 38041 address validation

### PR DESCRIPTION
**Pull request by acovrig**
## Summary
- Make `state` not required if country is not USA
- Make `internationalPostalCode` required if country is not USA
- Make `zipFirstFive` not required if country is not USA

## Related issue(s)
- [API-38041](https://jira.devops.va.gov/browse/API-38041)

## Testing done
- [x] *New code is covered by unit tests*
- Tested in postman and `rspec`
- International Change:
```
{ "data": { "type": "form/526", "attributes": { "claimProcessType": "STANDARD_CLAIM_PROCESS", "claimNotes": "Some things that are important to know, and are not included in any other place.", "veteranIdentification": { "serviceNumber": "123456789", "veteranNumber": { "telephone": "5555555555" }, "mailingAddress": { "addressLine1": "123 Main Street", "addressLine2": "Unit 1", "addressLine3": "Room 2", "city": "Portland", "state": "GA", "country": "Brazil", "internationalPostalCode": "asdf1234" }, "emailAddress": { "email": "valid@somedomain.com", "agreeToEmailRelatedToClaim": true }, "currentVaEmployee": false }, "changeOfAddress": { "typeOfAddressChange": "TEMPORARY", "addressLine1": "456 Main Street", "addressLine2": "Unit 3", "addressLine3": "Room 4", "city": "Atlanta", "state": "GA", "country": "Brazil", "internationalPostalCode": "qwerty1234", "dates": { "beginDate": "2025-06-04", "endDate": "2026-06-04" } }, "homeless": { "currentlyHomeless": { "homelessSituationOptions": "OTHER", "otherDescription": "Other living situation" }, "pointOfContact": "John Doe", "pointOfContactNumber": { "telephone": "5555555555" } }, "toxicExposure": { "gulfWarHazardService": { "servedInGulfWarHazardLocations": "NO" }, "herbicideHazardService": { "servedInHerbicideHazardLocations": "YES", "otherLocationsServed": "Other locations served", "serviceDates": { "beginDate": "1972-05", "endDate": "1980-10" } }, "additionalHazardExposures": { "additionalExposures": [ "OTHER" ], "specifyOtherExposures": "Agent Orange", "exposureDates": { "beginDate": "1972-05", "endDate": "1980-10" } }, "multipleExposures":[ { "hazardExposedTo":"Agent Orange", "exposureLocation":"Vietnam", "exposureDates": { "beginDate":"1972-05", "endDate":"1973-01" } }, { "hazardExposedTo":"Agent Orange", "exposureLocation":"Vietnam", "exposureDates": { "beginDate":"1979-04", "endDate":"1980-10" } } ] }, "disabilities": [ { "name": "Diabetes", "exposureOrEventOrInjury": "Agent Orange", "serviceRelevance": "Service in Vietnam War", "approximateDate": "1975-05", "disabilityActionType": "NEW", "isRelatedToToxicExposure": true }, { "name":"Hearing Loss", "exposureOrEventOrInjury":"Noise", "serviceRelevance":"Heavy equipment operator in service", "approximateDate":"1968-07", "disabilityActionType": "INCREASE", "classificationCode":"8987", "ratedDisabilityId":"1234567", "diagnosticCode":5678, "isRelatedToToxicExposure":false } ], "treatments": [ { "beginDate": "2021-04", "treatedDisabilityNames": [ "Diabetes" ], "center": { "name": "ATLANTA VA MEDICAL CENTER", "state": "GA", "city": "ATLANTA" } }, { "beginDate": "1996-03", "treatedDisabilityNames": [ "Hearing Loss" ], "center": { "name": "ATLANTA VA MEDICAL CENTER", "state": "GA", "city": "ATLANTA" } } ], "serviceInformation": { "alternateNames":[ "Jon Doe", "Jane Doe" ], "servicePeriods": [ { "serviceBranch": "Air Force", "serviceComponent": "Active", "activeDutyBeginDate": "1964-11-14", "activeDutyEndDate": "1980-10-30", "separationLocationCode": "98289" } ], "servedInActiveCombatSince911": "NO", "reservesNationalGuardService": { "component": "National Guard", "obligationTermsOfService": { "beginDate":"1990-11-24", "endDate":"1995-11-17" }, "unitName": "National Guard Unit Name", "unitAddress": "1243 Main Street", "unitPhone": { "areaCode": "555", "phoneNumber": "5555555" }, "receivingInactiveDutyTrainingPay": "YES" }, "confinements": [ { "approximateBeginDate":"1970-06-11", "approximateEndDate":"1970-09-11" } ] }, "servicePay": { "receivingMilitaryRetiredPay": "NO", "futureMilitaryRetiredPay": "YES", "futureMilitaryRetiredPayExplanation": "Explanation of future military retired pay", "militaryRetiredPay": { "branchOfService": "Air Force", "monthlyAmount": 240 }, "retiredStatus": "PERMANENT_DISABILITY_RETIRED_LIST", "favorMilitaryRetiredPay": false, "receivedSeparationOrSeverancePay": "YES", "separationSeverancePay": { "datePaymentReceived": "2018-07-31", "branchOfService": "Air Force", "preTaxAmountReceived": 179 }, "favorTrainingPay": false }, "directDeposit": { "accountNumber": "123123123123", "accountType": "CHECKING", "financialInstitutionName": "Chase", "routingNumber": "123456789" }, "claimantCertification": true } } }
```
- US change:
```{ "data": { "type": "form/526", "attributes": { "claimProcessType": "STANDARD_CLAIM_PROCESS", "claimNotes": "Some things that are important to know, and are not included in any other place.", "veteranIdentification": { "serviceNumber": "123456789", "veteranNumber": { "telephone": "5555555555" }, "mailingAddress": { "addressLine1": "123 Main Street", "addressLine2": "Unit 1", "addressLine3": "Room 2", "city": "Portland", "state": "GA", "country": "Brazil", "internationalPostalCode": "asdf1234" }, "emailAddress": { "email": "valid@somedomain.com", "agreeToEmailRelatedToClaim": true }, "currentVaEmployee": false }, "changeOfAddress": { "typeOfAddressChange": "TEMPORARY", "addressLine1": "456 Main Street", "addressLine2": "Unit 3", "addressLine3": "Room 4", "city": "Atlanta", "state": "GA", "country": "USA", "zipFirstFive": "42220", "zipLastFour": "9897", "dates": { "beginDate": "2025-06-04", "endDate": "2026-06-04" } }, "homeless": { "currentlyHomeless": { "homelessSituationOptions": "OTHER", "otherDescription": "Other living situation" }, "pointOfContact": "John Doe", "pointOfContactNumber": { "telephone": "5555555555" } }, "toxicExposure": { "gulfWarHazardService": { "servedInGulfWarHazardLocations": "NO" }, "herbicideHazardService": { "servedInHerbicideHazardLocations": "YES", "otherLocationsServed": "Other locations served", "serviceDates": { "beginDate": "1972-05", "endDate": "1980-10" } }, "additionalHazardExposures": { "additionalExposures": [ "OTHER" ], "specifyOtherExposures": "Agent Orange", "exposureDates": { "beginDate": "1972-05", "endDate": "1980-10" } }, "multipleExposures":[ { "hazardExposedTo":"Agent Orange", "exposureLocation":"Vietnam", "exposureDates": { "beginDate":"1972-05", "endDate":"1973-01" } }, { "hazardExposedTo":"Agent Orange", "exposureLocation":"Vietnam", "exposureDates": { "beginDate":"1979-04", "endDate":"1980-10" } } ] }, "disabilities": [ { "name": "Diabetes", "exposureOrEventOrInjury": "Agent Orange", "serviceRelevance": "Service in Vietnam War", "approximateDate": "1975-05", "disabilityActionType": "NEW", "isRelatedToToxicExposure": true }, { "name":"Hearing Loss", "exposureOrEventOrInjury":"Noise", "serviceRelevance":"Heavy equipment operator in service", "approximateDate":"1968-07", "disabilityActionType": "INCREASE", "classificationCode":"8987", "ratedDisabilityId":"1234567", "diagnosticCode":5678, "isRelatedToToxicExposure":false } ], "treatments": [ { "beginDate": "2021-04", "treatedDisabilityNames": [ "Diabetes" ], "center": { "name": "ATLANTA VA MEDICAL CENTER", "state": "GA", "city": "ATLANTA" } }, { "beginDate": "1996-03", "treatedDisabilityNames": [ "Hearing Loss" ], "center": { "name": "ATLANTA VA MEDICAL CENTER", "state": "GA", "city": "ATLANTA" } } ], "serviceInformation": { "alternateNames":[ "Jon Doe", "Jane Doe" ], "servicePeriods": [ { "serviceBranch": "Air Force", "serviceComponent": "Active", "activeDutyBeginDate": "1964-11-14", "activeDutyEndDate": "1980-10-30", "separationLocationCode": "98289" } ], "servedInActiveCombatSince911": "NO", "reservesNationalGuardService": { "component": "National Guard", "obligationTermsOfService": { "beginDate":"1990-11-24", "endDate":"1995-11-17" }, "unitName": "National Guard Unit Name", "unitAddress": "1243 Main Street", "unitPhone": { "areaCode": "555", "phoneNumber": "5555555" }, "receivingInactiveDutyTrainingPay": "YES" }, "confinements": [ { "approximateBeginDate":"1970-06-11", "approximateEndDate":"1970-09-11" } ] }, "servicePay": { "receivingMilitaryRetiredPay": "NO", "futureMilitaryRetiredPay": "YES", "futureMilitaryRetiredPayExplanation": "Explanation of future military retired pay", "militaryRetiredPay": { "branchOfService": "Air Force", "monthlyAmount": 240 }, "retiredStatus": "PERMANENT_DISABILITY_RETIRED_LIST", "favorMilitaryRetiredPay": false, "receivedSeparationOrSeverancePay": "YES", "separationSeverancePay": { "datePaymentReceived": "2018-07-31", "branchOfService": "Air Force", "preTaxAmountReceived": 179 }, "favorTrainingPay": false }, "directDeposit": { "accountNumber": "123123123123", "accountType": "CHECKING", "financialInstitutionName": "Chase", "routingNumber": "123456789" }, "claimantCertification": true } } }
```
- International Current:
```{ "data": { "type": "form/526", "attributes": { "claimProcessType": "STANDARD_CLAIM_PROCESS", "claimNotes": "Some things that are important to know, and are not included in any other place.", "veteranIdentification": { "serviceNumber": "123456789", "veteranNumber": { "telephone": "5555555555" }, "mailingAddress": { "addressLine1": "123 Main Street", "addressLine2": "Unit 1", "addressLine3": "Room 2", "city": "Portland", "state": "GA", "country": "Brazil", "internationalPostalCode": "asdf1234" }, "emailAddress": { "email": "valid@somedomain.com", "agreeToEmailRelatedToClaim": true }, "currentVaEmployee": false }, "changeOfAddress": { "typeOfAddressChange": "TEMPORARY", "addressLine1": "456 Main Street", "addressLine2": "Unit 3", "addressLine3": "Room 4", "city": "Atlanta", "state": "GA", "country": "USA", "zipFirstFive": "42220", "zipLastFour": "9897", "dates": { "beginDate": "2025-06-04", "endDate": "2026-06-04" } }, "homeless": { "currentlyHomeless": { "homelessSituationOptions": "OTHER", "otherDescription": "Other living situation" }, "pointOfContact": "John Doe", "pointOfContactNumber": { "telephone": "5555555555" } }, "toxicExposure": { "gulfWarHazardService": { "servedInGulfWarHazardLocations": "NO" }, "herbicideHazardService": { "servedInHerbicideHazardLocations": "YES", "otherLocationsServed": "Other locations served", "serviceDates": { "beginDate": "1972-05", "endDate": "1980-10" } }, "additionalHazardExposures": { "additionalExposures": [ "OTHER" ], "specifyOtherExposures": "Agent Orange", "exposureDates": { "beginDate": "1972-05", "endDate": "1980-10" } }, "multipleExposures":[ { "hazardExposedTo":"Agent Orange", "exposureLocation":"Vietnam", "exposureDates": { "beginDate":"1972-05", "endDate":"1973-01" } }, { "hazardExposedTo":"Agent Orange", "exposureLocation":"Vietnam", "exposureDates": { "beginDate":"1979-04", "endDate":"1980-10" } } ] }, "disabilities": [ { "name": "Diabetes", "exposureOrEventOrInjury": "Agent Orange", "serviceRelevance": "Service in Vietnam War", "approximateDate": "1975-05", "disabilityActionType": "NEW", "isRelatedToToxicExposure": true }, { "name":"Hearing Loss", "exposureOrEventOrInjury":"Noise", "serviceRelevance":"Heavy equipment operator in service", "approximateDate":"1968-07", "disabilityActionType": "INCREASE", "classificationCode":"8987", "ratedDisabilityId":"1234567", "diagnosticCode":5678, "isRelatedToToxicExposure":false } ], "treatments": [ { "beginDate": "2021-04", "treatedDisabilityNames": [ "Diabetes" ], "center": { "name": "ATLANTA VA MEDICAL CENTER", "state": "GA", "city": "ATLANTA" } }, { "beginDate": "1996-03", "treatedDisabilityNames": [ "Hearing Loss" ], "center": { "name": "ATLANTA VA MEDICAL CENTER", "state": "GA", "city": "ATLANTA" } } ], "serviceInformation": { "alternateNames":[ "Jon Doe", "Jane Doe" ], "servicePeriods": [ { "serviceBranch": "Air Force", "serviceComponent": "Active", "activeDutyBeginDate": "1964-11-14", "activeDutyEndDate": "1980-10-30", "separationLocationCode": "98289" } ], "servedInActiveCombatSince911": "NO", "reservesNationalGuardService": { "component": "National Guard", "obligationTermsOfService": { "beginDate":"1990-11-24", "endDate":"1995-11-17" }, "unitName": "National Guard Unit Name", "unitAddress": "1243 Main Street", "unitPhone": { "areaCode": "555", "phoneNumber": "5555555" }, "receivingInactiveDutyTrainingPay": "YES" }, "confinements": [ { "approximateBeginDate":"1970-06-11", "approximateEndDate":"1970-09-11" } ] }, "servicePay": { "receivingMilitaryRetiredPay": "NO", "futureMilitaryRetiredPay": "YES", "futureMilitaryRetiredPayExplanation": "Explanation of future military retired pay", "militaryRetiredPay": { "branchOfService": "Air Force", "monthlyAmount": 240 }, "retiredStatus": "PERMANENT_DISABILITY_RETIRED_LIST", "favorMilitaryRetiredPay": false, "receivedSeparationOrSeverancePay": "YES", "separationSeverancePay": { "datePaymentReceived": "2018-07-31", "branchOfService": "Air Force", "preTaxAmountReceived": 179 }, "favorTrainingPay": false }, "directDeposit": { "accountNumber": "123123123123", "accountType": "CHECKING", "financialInstitutionName": "Chase", "routingNumber": "123456789" }, "claimantCertification": true } } }
```
- US Current:
```{ "data": { "type": "form/526", "attributes": { "claimProcessType": "STANDARD_CLAIM_PROCESS", "claimNotes": "Some things that are important to know, and are not included in any other place.", "veteranIdentification": { "serviceNumber": "123456789", "veteranNumber": { "telephone": "5555555555" }, "mailingAddress": { "addressLine1": "123 Main Street", "addressLine2": "Unit 1", "addressLine3": "Room 2", "city": "Portland", "country": "USA", "state": "GA", "zipFirstFive": "12345", "zipLastFour": "1234" }, "emailAddress": { "email": "valid@somedomain.com", "agreeToEmailRelatedToClaim": true }, "currentVaEmployee": false }, "changeOfAddress": { "typeOfAddressChange": "TEMPORARY", "addressLine1": "456 Main Street", "addressLine2": "Unit 3", "addressLine3": "Room 4", "city": "Atlanta", "state": "GA", "country": "USA", "zipFirstFive": "42220", "zipLastFour": "9897", "dates": { "beginDate": "2025-06-04", "endDate": "2026-06-04" } }, "homeless": { "currentlyHomeless": { "homelessSituationOptions": "OTHER", "otherDescription": "Other living situation" }, "pointOfContact": "John Doe", "pointOfContactNumber": { "telephone": "5555555555" } }, "toxicExposure": { "gulfWarHazardService": { "servedInGulfWarHazardLocations": "NO" }, "herbicideHazardService": { "servedInHerbicideHazardLocations": "YES", "otherLocationsServed": "Other locations served", "serviceDates": { "beginDate": "1972-05", "endDate": "1980-10" } }, "additionalHazardExposures": { "additionalExposures": [ "OTHER" ], "specifyOtherExposures": "Agent Orange", "exposureDates": { "beginDate": "1972-05", "endDate": "1980-10" } }, "multipleExposures":[ { "hazardExposedTo":"Agent Orange", "exposureLocation":"Vietnam", "exposureDates": { "beginDate":"1972-05", "endDate":"1973-01" } }, { "hazardExposedTo":"Agent Orange", "exposureLocation":"Vietnam", "exposureDates": { "beginDate":"1979-04", "endDate":"1980-10" } } ] }, "disabilities": [ { "name": "Diabetes", "exposureOrEventOrInjury": "Agent Orange", "serviceRelevance": "Service in Vietnam War", "approximateDate": "1975-05", "disabilityActionType": "NEW", "isRelatedToToxicExposure": true }, { "name":"Hearing Loss", "exposureOrEventOrInjury":"Noise", "serviceRelevance":"Heavy equipment operator in service", "approximateDate":"1968-07", "disabilityActionType": "INCREASE", "classificationCode":"8987", "ratedDisabilityId":"1234567", "diagnosticCode":5678, "isRelatedToToxicExposure":false } ], "treatments": [ { "beginDate": "2021-04", "treatedDisabilityNames": [ "Diabetes" ], "center": { "name": "ATLANTA VA MEDICAL CENTER", "state": "GA", "city": "ATLANTA" } }, { "beginDate": "1996-03", "treatedDisabilityNames": [ "Hearing Loss" ], "center": { "name": "ATLANTA VA MEDICAL CENTER", "state": "GA", "city": "ATLANTA" } } ], "serviceInformation": { "alternateNames":[ "Jon Doe", "Jane Doe" ], "servicePeriods": [ { "serviceBranch": "Air Force", "serviceComponent": "Active", "activeDutyBeginDate": "1964-11-14", "activeDutyEndDate": "1980-10-30", "separationLocationCode": "98289" } ], "servedInActiveCombatSince911": "NO", "reservesNationalGuardService": { "component": "National Guard", "obligationTermsOfService": { "beginDate":"1990-11-24", "endDate":"1995-11-17" }, "unitName": "National Guard Unit Name", "unitAddress": "1243 Main Street", "unitPhone": { "areaCode": "555", "phoneNumber": "5555555" }, "receivingInactiveDutyTrainingPay": "YES" }, "confinements": [ { "approximateBeginDate":"1970-06-11", "approximateEndDate":"1970-09-11" } ] }, "servicePay": { "receivingMilitaryRetiredPay": "NO", "futureMilitaryRetiredPay": "YES", "futureMilitaryRetiredPayExplanation": "Explanation of future military retired pay", "militaryRetiredPay": { "branchOfService": "Air Force", "monthlyAmount": 240 }, "retiredStatus": "PERMANENT_DISABILITY_RETIRED_LIST", "favorMilitaryRetiredPay": false, "receivedSeparationOrSeverancePay": "YES", "separationSeverancePay": { "datePaymentReceived": "2018-07-31", "branchOfService": "Air Force", "preTaxAmountReceived": 179 }, "favorTrainingPay": false }, "directDeposit": { "accountNumber": "123123123123", "accountType": "CHECKING", "financialInstitutionName": "Chase", "routingNumber": "123456789" }, "claimantCertification": true } } }
```

## What areas of the site does it impact?
V2 526 submit and validate

## Acceptance criteria
- [x]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.
- [x]  Events are being sent to the appropriate logging solution
- [x]  Documentation has been updated (link to documentation)
- [ ]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

### CODE REVIEW

File: modules/claims_api/app/swagger/claims_api/v2/dev/swagger.json
Comment (FonzMP) [inline modules/claims_api/app/swagger/claims_api/v2/dev/swagger.json:2839]: should we add a description to this specifically stating `if not 'USA'` just to clarify?

**Review by FonzMP — DISMISSED** [review]
Description change isn't a real "requirement" but think it's worth noting the description on internationalPostalCode. but just to unblock in case we decide not to. 

:shipit:

File: modules/claims_api/app/controllers/concerns/claims_api/v2/disability_compensation_validation.rb
Comment (acovrig) [reply modules/claims_api/app/controllers/concerns/claims_api/v2/disability_compensation_validation.rb:178]: We do a `form_attributes['']` before that and it'd fail there if `form_attributes` is `nil` however, `form_attributes` has a `|| {}` in [base.rb](https://github.com/department-of-veterans-affairs/vets-api/blob/master/modules/claims_api/app/controllers/claims_api/v2/veterans/base.rb#L23) in this case, so I presume it's fine w/out the `&`

**Review by rockwellwindsor-va — COMMENTED** [review]
@acovrig  The form does accept these values as stated in the ticket but I'm not sure the PDF generator does.  I am getting this returned.

Screenshot 2024-07-23 at 9 12 45 AM

This is what I sent in for the mailing Address, let me know if something looks off here (either way it was accepted so something likely needs to be tweaked)

```
     "veteranIdentification":{
      "currentVaEmployee": false,
      "mailingAddress":{
        "addressLine1":"BB Carpet Street",
        "city":"Portland",
        "country":"Brazil",
        "internationalPostalCode": "213"
      }
     },
```

**Comment by FonzMP** [comment]
> @acovrig The form does accept these values as stated in the ticket but I'm not sure the PDF generator does. I am getting this returned.
> 
> Screenshot 2024-07-23 at 9 12 45 AM
> 
> This is what I sent in for the mailing Address, **_let me know if something looks off here_** (either way it was accepted so something likely needs to be tweaked)
> 
> ```
>      "veteranIdentification":{
>       "currentVaEmployee": false,
>       "mailingAddress":{
>         "addressLine1":"BB Carpet Street",
>         "city":"Portland",
>         "country":"Brazil",
>         "internationalPostalCode": "213"
>       }
>      },
> ```
> 
> If that is the case wondering if this would be something that would need to be updated in the PDF generator so it was accepted and then added to overflow?

Good call out @rockwellwindsor-va . I think we have three options we'll need to explore.
1. Having Firefly add a specific input for it, but it doesn't really align with anywhere on the form that I know of, so probably not this one unless they just push it to the overflow box - which could be achieved by the below.
2. Provide it to the overflowText with a key to indicate? 
3. Ignore sending it to the PDF generator. 

cc: @klwatanabe

File: modules/claims_api/app/controllers/concerns/claims_api/v2/disability_compensation_validation.rb
Comment (mchristiansonVA) [inline modules/claims_api/app/controllers/concerns/claims_api/v2/disability_compensation_validation.rb:150]: Minor: I believe blank? also works with nil values, so could reduce to:
`return if form_attributes['veteranIdentification'].blank?`

**Review by rockwellwindsor-va — COMMENTED** [review]
@acovrig  Can we add a test for this internationalPostalCode and country as well?  Even just something like

```
      it 'maps veteran info correctly with an international country and postal code' do
        form_attributes['veteranIdentification']['mailingAddress']['country'] = 'Brazil'
        form_attributes['veteranIdentification']['mailingAddress']['internationalPostalCode'] = '5555'
        form_attributes['veteranIdentification']['mailingAddress']['zipFirstFive'] = nil
        form_attributes['veteranIdentification']['mailingAddress']['zipLastFour'] = nil
        mapper.map_claim

        mapped_zip = pdf_data[:data][:attributes][:identificationInformation][:mailingAddress][:zip]
        mapped_country = pdf_data[:data][:attributes][:identificationInformation][:mailingAddress][:country]
        expect(mapped_zip).to eq('5555')
        expect(mapped_country).to eq('Brazil')
      end
```

I think would be good (or whatever you feel like s worth locking in)

**Review by mchristiansonVA — COMMENTED** [review]
Getting a similar error to Rocky's for the change of address section with internationalPostalCode:
Screenshot 2024-07-24 at 2 17 15 PM

This was the request data for that:
```
      "changeOfAddress": {
        "typeOfAddressChange": "TEMPORARY",
        "addressLine1": "10 Peach St",
        "addressLine2": "Unit 4",
        "addressLine3": "Room 1",
        "city": "Atlanta",
        "country": "United Kingdom",
        "internationalPostalCode": "EH21 6UU",
        "dates": {
          "beginDate": "2023-06-04",
          "endDate": "2023-12-04"
        }
      }
```

**Review by mchristiansonVA — COMMENTED** [review]
Also getting this error from the EVSS docker container:
```
2024-07-24 14:20:07.560895 I [29205:puma srv tp 004] Rails -- ClaimsApi :: 526_docker_container :: {"detail":"EVSS DOCKER CONTAINER validate error: {:messages=>[{:key=>\"form526.veteran.currentMailingAddress.DomesticAddress\", :severity=>\"ERROR\", :text=>\"city, state and zipCode may not be null\"}]}","claim":"7135fc94-c768-4746-b627-611f0b447355"} :: Location: /Users/matthewchristianson/development/vets-api/modules/claims_api/lib/evss_service/base.rb:93
2024-07-24 14:20:07.568218 I [29205:puma srv tp 004] Rails -- Completed 422 Unprocessable Entity in 81860ms (Allocations: 231559)
```

Not sure how EVSS determines it's a domestic address? The request data for mailing address that resulted in the error:
```
        "mailingAddress": {
          "addressLine1": "1234 Couch Street",
          "addressLine2": "Unit 4",
          "addressLine3": "Room 1",
          "city": "Portland",
          "country": "United Kingdom",
          "internationalPostalCode": "EH21 6UU"
        }
```

**Comment by FonzMP** [comment]
> Also getting this error from the EVSS docker container:
> 
> ```
> 2024-07-24 14:20:07.560895 I [29205:puma srv tp 004] Rails -- ClaimsApi :: 526_docker_container :: {"detail":"EVSS DOCKER CONTAINER validate error: {:messages=>[{:key=>\"form526.veteran.currentMailingAddress.DomesticAddress\", :severity=>\"ERROR\", :text=>\"city, state and zipCode may not be null\"}]}","claim":"7135fc94-c768-4746-b627-611f0b447355"} :: Location: /Users/matthewchristianson/development/vets-api/modules/claims_api/lib/evss_service/base.rb:93
> 2024-07-24 14:20:07.568218 I [29205:puma srv tp 004] Rails -- Completed 422 Unprocessable Entity in 81860ms (Allocations: 231559)
> ```
> 
> Not sure how EVSS determines it's a domestic address? The request data for mailing address that resulted in the error:
> 
> ```
>         "mailingAddress": {
>           "addressLine1": "1234 Couch Street",
>           "addressLine2": "Unit 4",
>           "addressLine3": "Room 1",
>           "city": "Portland",
>           "country": "United Kingdom",
>           "internationalPostalCode": "EH21 6UU"
>         }
> ```

thinking this may come in to play: 
```
      type:
        type: "string"
        enum:
        - "DOMESTIC"
        - "MILITARY"
        - "INTERNATIONAL"
```

within the type definition of mailingAddress is specifically calls out a type, so it must **_assume_** Domestic. We may have to try submitting `INTERNATIONAL` if it gets the internationalPostalCode.

File: modules/claims_api/lib/claims_api/v2/disability_compensation_pdf_mapper.rb
Comment (stiehlrod) [inline modules/claims_api/lib/claims_api/v2/disability_compensation_pdf_mapper.rb:394]: @acovrig Since we have the zip logic in two places it would be nice to abstract it. Maybe another PR after this one is merged, since we are on a deadline.

**Comment by stiehlrod** [comment]
@acovrig It seems the state should not be required, correct?
Screenshot 2024-07-25 at 9 21 04 AM

**Comment by acovrig** [comment]
> @acovrig It seems the state should not be required, correct? Screenshot 2024-07-25 at 9 21 04 AM
> 
> Screenshot 2024-07-25 at 9 26 55 AM

* Interesting, state isn't required on our side, I wonder if we should set it to `" "` (empty space) if it's international?
* We don't have `type` on our schema, my last change sets it to `INTERNATIONAL` if the `internationalPostalCode` is set.

**Comment by stiehlrod** [comment]
@acovrig 
> * We don't have `type` on our schema, my last change sets it to `INTERNATIONAL` if the `internationalPostalCode` is set.

yep, was about to delete that screenshot, just trying things quick before standup that i saw in the comments

**Review by mchristiansonVA — COMMENTED** [review]
I tested setting [this line](https://github.com/department-of-veterans-affairs/vets-api/blob/7776f930c5eda85aead89b391d49664b3434f6d3/modules/claims_api/lib/claims_api/v2/disability_compensation_evss_mapper.rb#L54) to "INTERNATIONAL" and it ran successfully with no errors from the Docker container.

**Comment by stiehlrod** [comment]
> I tested setting the `type` on [this line](https://github.com/department-of-veterans-affairs/vets-api/blob/7776f930c5eda85aead89b391d49664b3434f6d3/modules/claims_api/lib/claims_api/v2/disability_compensation_evss_mapper.rb#L54) to "INTERNATIONAL" and it ran successfully with no errors from the Docker container.

Interesting, I wonder why that is hardcoded. Seems like a good find.

**Comment by stiehlrod** [comment]
@acovrig When I send in an address with both a zipFirstFive & internationalPostalCode, i get a 400 internal service error, and I feel like we should catch that with validation.

**Comment by klwatanabe** [comment]
> @acovrig When I send in an address with both a zipFirstFive & internationalPostalCode, i get a 400 internal service error, and I feel like we should catch that with validation. @klwatanabe is this something we want to address in this ticket, or start a new one so we can get this one in?

@acovrig - I think that should be covered in this story, but I'm okay with that being a separate sub-task if you prefer to get this piece merged first. Just don't want to close out the story until we have all of the error handling in place.

**Review by mchristiansonVA — APPROVED** [review]
Tested 526/validate and 526/synchronous with various address-related variations for mailingAddress and changeOfAddress; got successful requests using US and non-US address information, and validation as expected. LGTM!

**Pull request merged**
